objects.proto

```diff
@@ -6,22 +6,108 @@ option java_multiple_files = false;
 option java_package = "EcdarProtoBuf";
 option java_outer_classname = "ObjectProtos";
 
-message Zone {
-  int32 dimensions = 1;
-  repeated int32 matrix = 2;
-  // Maybe include a lookup table for the indices?
+import "component.proto";
+
+message ComponentInstance {
+  string component_name = 1;
+  uint32 component_index = 2;
+}
+
+message Clock {
+  message ComponentClock {
+    ComponentInstance component_instance = 1;
+    string clock_name = 2;
+  }
+  message SystemClock {
+    uint32 clock_index = 1;
+  }
+  message Zero {}
+  oneof clock {
+    // When a clock is attached to a component
+    ComponentClock component_clock = 1;
+    // When a clock is not attached to a component
+    SystemClock system_clock = 2;
+    // The zero clock
+    Zero zero = 3;
+  }
+}
+
+
+message Constraint {
+  // This message represents if (strict) { x - y < c } else { x - y <= c }
+  Clock x = 1;
+  Clock y = 2;
+  bool strict = 3;
+  int32 c = 4;
+}
+
+message Conjunction {
+  repeated Constraint constraints = 1;
+}
+
+message Disjunction {
+  repeated Conjunction conjunctions = 1;
+}
+
+message Federation {
+  Disjunction disjunction = 1;
+}
```

Comment:
Is a `Federation` just a wrapper for a disjunction?

Reply:
Yes, I have changed it to use a disjunction instead now. It's also much nicer to just use a disjunction because it is more general, e.g. if we switch to CDDs in the future, they can still be represented by a disjunction.